ilastik/applets/dataSelection/opDataSelection.py

```diff
@@ -64,8 +64,15 @@ def __init__(self, file_path: str, base_dir: str):
 
 
 class UnsuitedAxistagsException(Exception):
-    def __init__(self, axistags, shape):
-        super().__init__(f"Axistags {axistags} don't fit data shape {shape}")
+    def __init__(self, axistags: AxisTags, shape):
+        if len(axistags) > len(shape):
+            problem = "Inconsistent metadata: The dataset reports dimensions for which it contains no data.\n"
+        else:
+            problem = "Inconsistent metadata: The dataset has more dimensions than it reports.\n"
+        super().__init__(
+            problem + f"""Reported axes: {", ".join([tag.key for tag in axistags])}\n"""
+            f"""Actual data dimensions: {", ".join([str(s) for s in shape])} (pixels/channels/timepoints)"""
```

Comment:
not sure about the triple quote strings...

```suggestion
            f"{problem}Reported axes: {", ".join([tag.key for tag in axistags])}\n"
            f"Actual data dimensions: {", ".join([str(s) for s in shape])} (pixels/channels/timepoints)"
```

Reply:
I had to convert the {}-enclosed quotes to single quotes for this one but yeah, nicer this way.